bundler/lib/bundler/templates/newgem/lib/newgem.rb.tt

```diff
@@ -2,7 +2,14 @@
 
 require_relative "<%= File.basename(config[:namespaced_path]) %>/version"
 <%- if config[:ext] -%>
-require_relative "<%= File.basename(config[:namespaced_path]) %>/<%= config[:underscored_name] %>"
+# Attempt to load a versioned extension based on the Ruby version.
+# Fall back to loading the non-versioned extension if version-specific loading fails.
+begin
+  RUBY_VERSION =~ /(\d+\.\d+)/
+  require "#{Regexp.last_match(1)}/<%= File.basename(config[:namespaced_path]) %>/<%= config[:underscored_name] %>"
```

Comment:
This is something users frequently stumble on when trying to precompile gems, so let's just use the recommended pattern by rake-compiler as a default: https://github.com/rake-compiler/rake-compiler?tab=readme-ov-file#cross-compilation---the-future-is-now